src/controllers/csi/provisioner/agent.go

```diff
@@ -53,7 +53,7 @@ func (updater *agentUpdater) updateAgent(installedVersion, tenantUUID string, pr
 	targetVersion := updater.getOneAgentVersionFromInstance()
 	targetDir := updater.path.AgentBinaryDirForVersion(tenantUUID, targetVersion)
 
-	if _, err := updater.fs.Stat(targetDir); os.IsNotExist(err) {
+	if _, err := updater.fs.Stat(targetDir); os.IsNotExist(err) || installedVersion == "" {
```

Comment:
nitpick: since there is now a second condition to the error handling, it would be better to assign `err` before the condition instead of inside it.
```suggestion
    _, err := updater.fs.Stat(targetDir)
	if os.IsNotExist(err) || installedVersion == "" {
```